haproxy/tests/with_mock.spec.js

```diff
@@ -0,0 +1,56 @@
+const { expect } = require('chai');
+
+const waitForService = async (url, maxAttempts = 10, delay = 100) => {
+  for (let attempt = 0; attempt < maxAttempts; attempt++) {
+    try {
+      const response = await fetch(url, { 
+        method: 'GET',
+        timeout: 5000
```

Comment:
afaik Request doesn't have a timeout property: https://developer.mozilla.org/en-US/docs/Web/API/Request
You need to use an AbortSignal for this. 

Also, I think the default method will be GET. You don't need to specify it.